packages/react-ui-ag/src/Modals/MoreOptionsModal.js

```diff
@@ -0,0 +1,266 @@
+    isOpen: PropTypes.bool,
+    theme: PropTypes.object,
+    moreOptionsForm: PropTypes.node,
+    total: PropTypes.number,
+    filterTotal: PropTypes.number,
+    onClose: PropTypes.func,
+    onSubmit: PropTypes.func,
+    restoreFilters: PropTypes.func,
+    clearFilters: PropTypes.func,
+    applyFilters: PropTypes.func,
+  }
+
+  static defaultProps = {
+    theme: {},
+    total: 0,
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isOpen: true,
+    }
+  }
+
+  @autobind
+  getShowPropertyButton() {
+    const { filterTotal, total } = this.props
+
+    let count = (filterTotal !== null) ? filterTotal : total
+    let propertyCount = 'Show 0 Properties'
+
+    if (count > 1) {
+      count = this.formatNumberWithCommas(count)
+      propertyCount = `Show ${count} Properties`
+    } else if (count === 1) {
+      propertyCount = 'Show 1 Property'
+    }
+
+    return propertyCount
+  }
+
+  @autobind
+  getPropertyCount() {
+    const { filterTotal, total } = this.props
+    let count = (filterTotal !== null) ? filterTotal : total
+    let propertyCount = '0 Properties Found'
+
+    if (count > 1) {
+      count = this.formatNumberWithCommas(count)
+      propertyCount = `${count} Properties Found`
+    } else if (count === 1) {
+      propertyCount = '1 Property Found'
+    }
+
+    return propertyCount
+  }
+  @autobind
+  handleOutsideMouseDown(e) {
+    // If a click starts outside the modal, set the outside flag
+    // so when the click event occurs on the overlay,
+    // it will close the modal.
+    this.clickedOutside = (e.target === this.overlay)
+  }
+
+  @autobind
+  handleOutsideMouseUp(e) {
+    // If a click ends inside the modal, unset the outside flag
+    // so when the click event bubbles up to the overlay,
+    // it will not close the modal.
+    if (e.target !== this.overlay) {
+      this.clickedOutside = false
+    }
+  }
+
+  @autobind
+  handleOutsideClick(e) {
+    const { onClose, applyFilters } = this.props
+
+    // Verify the click started on the overlay.
+    // This will handle the case of the user starting a click inside the modal,
+    // then dragging and releasing the button outside the modal.
+    if (!this.clickedOutside) {
+      return
+    }
+
+    // Clear the flag so it will not affect the next click
+    this.clickedOutside = null
+
+    // Verify the click ended on the overlay and did not bubble up
+    // from the modal
+    if (e.target !== this.overlay) {
+      return
+    }
+
+    if (onClose) {
+      onClose()
+      if (applyFilters) {
+        applyFilters()
+      }
+    }
+  }
+
+  @autobind
+  handleSubmit(e) {
+    const {
+      onSubmit,
+      onClose,
+    } = this.props
+
+    if (!onSubmit) {
+      e.preventDefault()
+    } else {
+      onSubmit()
+    }
+
+    if (onClose) {
+      onClose()
+    }
+  }
+
+  @autobind
+  handleCancel() {
+    const {
+      onClose,
+      restoreFilters,
+    } = this.props
+    restoreFilters()
```

Comment:
This is overkill. No reason to do this here. Why can’t it be done in onClose?

Reply:
@tadjohnston On close will be called whenever the modal closes, so even onSubmit, so putting restorefilters inside of it would not work all the time.

Reply:
So we don't reset the filters on submit, but we do on cancel? I would think they'd do the same thing if they both close the modal.

Reply:
@tadjohnston from my understanding of the functionality,  cancel is used to restore filters to the way they were when you first entered the modal and then close the modal, reset all is used to clear all filters, but the modal stays open and submit is used to apply the filters in the modal and close the modal. However I changed the prop from onClose to closeModal because it is a better description of how it is being used. To integrate, with ModalStack I have to pass in a function that will dispatch an action to close the modal via modal stack. So yes we don't restoreFilters on submit, but we do on cancel.